jwql/website/apps/jwql/forms.py

```diff
@@ -380,3 +532,19 @@ class MnemonicExplorationForm(forms.Form):
                                    help_text="String ID (tlmMnemonic)")
     unit = forms.CharField(label='unit', max_length=500, required=False,
                            help_text="unit")
+
+
+class ObservingModeForm(forms.Form): # Add instruments chosen parameter
+    """Creates a ``ObservingModeForm`` object that allows for ``mode``
+    input in a form field.
+    """
+    modelist=[]
+    for instrument in OBSERVING_MODE_PER_INSTRUMENT.keys():  # Add AND in instruments chosen
+        modes_per_inst = OBSERVING_MODE_PER_INSTRUMENT[instrument]
+        for mode in modes_per_inst:
+            modelist.append([mode, mode]) if [mode, mode] not in modelist else modelist
+    mode = forms.MultipleChoiceField(required=False, choices=modelist, widget=forms.CheckboxSelectMultiple)
+
+    def clean_modes(self):
+        modes=self.cleaned_data['mode']
+        return modes
```

Comment:
Great work on adding all of these forms!  I'm still wondering if we should be using one gigantic form, or many smaller ones as you have here ... Something that we can research and figure out for the next iteration updates.

Reply:
Thank you! I'm working on adding in the multi-widget form; I'll email you with more detailed updates on that.